src/core/AutoRest.Core/Validation/Rule.cs

```diff
@@ -16,9 +16,15 @@ protected Rule()
         {
         }
 
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
```

Comment:
A C# newbie question here: would it make sense to declare Id as an abstract property so any subclass Must have its own Id?